remotes/docker/resolver.go

```diff
@@ -428,13 +435,15 @@ func (r *dockerBase) setTokenAuth(ctx context.Context, params map[string]string)
 		service: params["service"],
 	}
 
-	scope, ok := params["scope"]
-	if !ok {
-		return errors.Errorf("no scope specified for token auth challenge")
+	if x := ctx.Value(tokenScopesKey{}); x != nil {
```

Comment:
I think it is fine to pass the additional scope through the context like this. However I don't think we should blindly append to the scope list. Since right now `params["scope"]` is only returning one element, I would compare those in the context against that scope to make sure we don't duplicate the value. I think this could be important for caching purposes as I believe we will want to cache this in the future by scope.